test/unit/services/sshAdapter_spec.js

```diff
@@ -0,0 +1,357 @@
+"use strict";
+
+var AWS        = require("aws-sdk");
+var sinon      = require("sinon");
+var expect     = require("chai").expect;
+var SshAdapter = require("../../../lib/services/sshAdapter");
+var Bluebird   = require("bluebird");
+var options    = {};
+var ec2        = new AWS.EC2();
+
+Bluebird.promisifyAll(ec2);
+options.ec2 =  ec2;
+
+describe("The SSH Adapter Class", function () {
```

Comment:
This test suite seems to only verify the behavior of the initial connection attempt. Unless I missed something, I didn't see any assertions validating that polling continues when SSH is not yet ready. Should we also support and validate a timeout if the instance does not fail but SSH also does not connect within a certain amount of time?